roles/st2smoketests/tasks/main.yml

```diff
@@ -49,3 +49,102 @@
   changed_when: no
   tags:
     - smoke-tests
+
+- name: Generate st2api key
+  command: st2 apikey create -k
+  environment:
+    ST2_AUTH_TOKEN: "{{ st2_token.stdout }}"
+  register: st2_api_key
+  changed_when: no
+  no_log: true
+  tags:
+    - smoke-tests
+
+- name: Set st2chatops_api_key
+  set_fact:
+    st2chatops_api_key: "{{ st2_api_key.stdout  }}"
+    st2chatops_hubot_adapter: "slack"
+    st2chatops_config: {"HUBOT_SLACK_TOKEN": "{{ hubot_token }}"}
+  changed_when: no
+  no_log: true
+  tags:
+    - smoke-tests
+
+- name: Install st2 pack from exchange
+  become: yes
+  environment:
+    ST2_AUTH_TOKEN: "{{ st2_token.stdout }}"
+  command: "st2 pack install st2"
+  changed_when: no
+  tags:
+    - smoke-tests
+
+- name: Update st2_api_key in st2chatops.env
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '(?<=ST2_API_KEY=)(.*)$'
+    replace: '{{ st2chatops_api_key }}'
+  no_log: true
+  changed_when: no
+  tags:
+    - smoke-tests
+
+- name: Comment Username, Password and Auth URL, if API_KEY provided
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '{{ item }}'
+    replace: '# \1'
+  with_items:
+    - '^(export ST2_AUTH_URL.*)'
+    - '^(export ST2_AUTH_USERNAME.*)'
+    - '^(export ST2_AUTH_PASSWORD.*)'
+  tags:
+    - smoke-tests
+
+- name: Comment existing hubot adapters in "/opt/stackstorm/chatops/st2chatops.env"
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '^(export HUBOT_ADAPTER=.*)'
+    replace: '# \1'
+  changed_when: no
+  tags:
+    - smoke-tests
+
+- name: Update "SLACK" adapter
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '^# (export HUBOT_ADAPTER={{ st2chatops_hubot_adapter }})$'
+    replace: '\1'
+  changed_when: no
+  tags:
+    - smoke-tests
+
+- name: Configure "SLACK" settings
+  become: yes
+  ini_file:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    section: null
+    option: 'export {{ _conf_option.key }}'
+    value: '{{ _conf_option.value }}'
+    no_extra_spaces: yes
+  with_dict: '{{ st2chatops_config }}'
+  loop_control:
+    loop_var: _conf_option
+  no_log: true
+  notify: restart st2chatops
+  tags:
+    - smoke-tests
+
+- meta: flush_handlers
+  tags:
+    - smoke-tests
+
+- name: Verify st2chatops using bin/hubot
+  shell: "cd /opt/stackstorm/chatops/ && { echo -n; sleep 5; echo '! st2 list actions'; echo; sleep 10; } | bin/hubot --test "
```

Comment:
With `bin/hubot --test` we actually test `shell` adapter.
See: https://github.com/StackStorm/st2chatops/blob/master/bin/hubot#L20-L22

So decide whether you test `slack` or `shell` hubot adapter.